tensorflow/cc/framework/gradients.cc

```diff
@@ -165,6 +194,9 @@ Status SymbolicGradientBuilder::Initialize() {
   // handle computing gradients over small subgraphs from a very large graph.
   pending_.resize(scope_.graph()->num_node_ids(), 0);
   {
+    std::vector<bool> reachable_nodes;
+    GetReachableNodes(&reachable_nodes);
+    
```

Comment:
I moved `std::vector<bool> reachable_nodes = GetReachableNodes();` at the beginning of the method as we'll check the errors at the beginning (the TODO).